vsintegration/src/FSharp.Editor/QuickInfo/QuickInfoProvider.fs

```diff
@@ -26,89 +25,127 @@ open Microsoft.VisualStudio.Language.Intellisense
 
-        let workspace = thisDoc.Project.Solution.Workspace
-        let documentNavigationService = workspace.Services.GetService<IDocumentNavigationService>()
+        let workspace = initialDoc.Project.Solution.Workspace
         let solution = workspace.CurrentSolution
 
-        let documentId (range: range) =
+
+        /// Retrieve the DocumentId from the workspace for the range's file
+        let docIdOfRange (range: range) =
             let filePath = System.IO.Path.GetFullPathSafe range.FileName
-            let projectOf (id : DocumentId) = solution.GetDocument(id).Project
 
             //The same file may be present in many projects. We choose one from current or referenced project.
             let rec matchingDoc = function
             | [] -> None
-            | id::_ when projectOf id = thisDoc.Project || IsScript thisDoc.FilePath -> Some id
-            | id::tail -> 
-                if (projectOf id).GetDependentProjects() |> Seq.contains thisDoc.Project then Some id
+            | (docId:DocumentId)::_ when docId.ProjectId = initialDoc.Id.ProjectId || IsScript initialDoc.FilePath -> Some docId
+            | docId::tail -> 
+                if solution.GetDependentProjectIds(docId.ProjectId).Contains initialDoc.Id.ProjectId then Some docId
                 else matchingDoc tail
-            solution.GetDocumentIdsWithFilePath(filePath) |> List.ofSeq |> matchingDoc
+            solution.GetDocumentIdsWithFilePath filePath |> List.ofSeq |> matchingDoc
+
 
         let canGoTo range =
-            range <> rangeStartup && documentId range |> Option.isSome
+            range <> rangeStartup && docIdOfRange range |> Option.isSome
+
+        // to ensure proper navigation decsions we need to check the type of document the navigation call
+        // is originating from and the target we're provided by default
+        //  - signature files (.fsi) should navigate to other signature files 
+        //  - implementation files (.fs) should navigate to other implementation files
+        //let adjustTarget (range:range) =
 
-        let goTo range = 
+        let navigateTo (range:range) = 
             asyncMaybe { 
-                let! id = documentId range
-                let! src = solution.GetDocument(id).GetTextAsync()
-                let! span = CommonRoslynHelpers.TryFSharpRangeToTextSpan(src, range)
-                if documentNavigationService.TryNavigateToSpan(workspace, id, span) then
-                   let! session = currentSession
-                   session.Dismiss()   
-            } |> Async.Ignore |> Async.StartImmediate 
-
-        let formatMap = typemap.ClassificationFormatMapService.GetClassificationFormatMap("tooltip")
-
-        let props = 
-            ClassificationTags.GetClassificationTypeName
+                let targetPath = range.FileName 
+                logInfof "origin document - %s \n"  initialDoc.FilePath
+                logInfof "target range to - %s \n %A"  range.FileName range
```

Comment:
Remove logging. It will appear in release configuration as well, which we certainly don't want.